File: 2019/01/01.jl
input = readlines()
@assert(length(input) > 0, "puzzle input must be passed to stdin.")

module_masses = parse.(Int64, input[1])

# part 1
fuel = sum(div.(module_masses, 3) .- 2)
println(fuel)

# part 2
calc_fuel_required = function(mass)
    fuel = div(mass, 3) - 2
    fuel <= 0 ? 0 : fuel + calc_fuel_required(fuel)
end

fuel = sum(calc_fuel_required.(module_masses))
println(fuel)

File: 2019/01/01.py
import os

assert os.path.isfile("data.txt"), "input must be saved in '2019/01/data.txt'"
with open("data.txt") as f:
    module_masses = [int(mass) for mass in f.readlines()]

# part 1
fuel = sum([mass // 3 - 2 for mass in module_masses])
print(f"Part 1 Fuel Required: {fuel}")

# part 2
def calc_fuel_required(mass):
    fuel = mass // 3 - 2
    return 0 if fuel <= 0 else fuel + calc_fuel_required(fuel)

fuel = sum([calc_fuel_required(mass) for mass in module_masses])
print(f"Part 2 Fuel Required: {fuel}")
